test/integration/filter_manager_integration_test.cc

```diff
@@ -521,6 +524,54 @@ TEST_P(InjectDataWithTcpProxyFilterIntegrationTest, UsageOfInjectDataMethodsShou
   tcp_client->waitForDisconnect();
 }
 
+class FilterChainAccessLogTest : public testing::TestWithParam<Network::Address::IpVersion>,
+                                 public BaseIntegrationTest {
+public:
+  explicit FilterChainAccessLogTest()
+      : BaseIntegrationTest(GetParam(), ConfigHelper::tcpProxyConfig()) {}
+};
+
+INSTANTIATE_TEST_SUITE_P(Params, FilterChainAccessLogTest,
+                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
+                         TestUtility::ipTestParamsToString);
+
+TEST_P(FilterChainAccessLogTest, FilterChainName) {
+  auto log_file = TestEnvironment::temporaryPath(TestUtility::uniqueFilename());
+
+  config_helper_.addConfigModifier([&](envoy::config::bootstrap::v3::Bootstrap& bootstrap) -> void {
+    auto* listener = bootstrap.mutable_static_resources()->mutable_listeners(0);
+    auto* filter_chain = listener->mutable_filter_chains(0);
+    filter_chain->set_name("foo_filter_chain");
+    auto* config_blob = filter_chain->mutable_filters(0)->mutable_typed_config();
+
+    auto tcp_proxy_config = MessageUtil::anyConvert<API_NO_BOOST(
+        envoy::extensions::filters::network::tcp_proxy::v3::TcpProxy)>(*config_blob);
+
+    auto* access_log = tcp_proxy_config.add_access_log();
+    access_log->set_name("accesslog");
+    envoy::extensions::access_loggers::file::v3::FileAccessLog access_log_config;
+    access_log_config.set_path(log_file);
+    access_log_config.mutable_log_format()->mutable_text_format_source()->set_inline_string(
+        "%RESPONSE_FLAGS% %FILTER_CHAIN_NAME%");
+    access_log->mutable_typed_config()->PackFrom(access_log_config);
+    config_blob->PackFrom(tcp_proxy_config);
+  });
+  enableHalfClose(true);
+  initialize();
+
+  auto tcp_client = makeTcpConnection(lookupPort("listener_0"));
+  ASSERT_TRUE(tcp_client->write("hello", true));
+
+  FakeRawConnectionPtr fake_upstream_connection;
+  ASSERT_TRUE(fake_upstreams_[0]->waitForRawConnection(fake_upstream_connection));
+  ASSERT_TRUE(fake_upstream_connection->waitForData(5));
+  ASSERT_TRUE(fake_upstream_connection->close());
+  tcp_client->close();
+  ASSERT_TRUE(fake_upstream_connection->waitForDisconnect());
+
+  EXPECT_THAT(waitForAccessLog(log_file), testing::HasSubstr("- foo_filter_chain"));
```

Comment:
Why is there a - in this case?  Comments above seemed to incidate that - was logged when no filter chain was selected.

Reply:
The dash is actually "%RESPONSE_FLAGS% 
I add another "-" to make sure foo_filter_chain I am not accidentally fooled by a log line somewher eles.